commands/azureCommands/acr-build.ts

```diff
@@ -55,35 +51,39 @@ export async function queueBuild(dockerFileUri?: vscode.Uri): Promise<void> {
     let tarFilePath = getTempSourceArchivePath();
 
     status.appendLine("Uploading Source Code to " + tarFilePath);
-    sourceLocation = await uploadSourceCode(client, registryName, resourceGroupName, sourceLocation, tarFilePath);
+    let uploadedSourceLocation = await uploadSourceCode(client, registry.name, resourceGroupName, sourceLocation, tarFilePath, folder);
 
     let osType = os.type()
     if (osType === 'Windows_NT') {
         osType = 'Windows'
     }
-
     status.appendLine("Setting up Build Request");
     let buildRequest: QuickBuildRequest = {
         'type': 'QuickBuild',
         'imageNames': [name],
         'isPushEnabled': true,
-        'sourceLocation': sourceLocation,
-        'platform': { 'osType': osType },
+        'sourceLocation': uploadedSourceLocation,
+        'platform': { 'osType': 'Linux' },
         'dockerFilePath': relativeDockerPath
     };
     status.appendLine("Queueing Build");
-    await client.registries.queueBuild(resourceGroupName, registryName, buildRequest);
+    await client.registries.queueBuild(resourceGroupName, registry.name, buildRequest);
     status.appendLine('Success');
 }
 
-async function uploadSourceCode(client: ContainerRegistryManagementClient, registryName: string, resourceGroupName: string, sourceLocation: string, tarFilePath: string): Promise<string> {
+async function uploadSourceCode(client: ContainerRegistryManagementClient, registryName: string, resourceGroupName: string, sourceLocation: string, tarFilePath: string, folder: vscode.WorkspaceFolder): Promise<string> {
     status.appendLine("   Sending source code to temp file");
-    tar.c(
-        {
-            gzip: true
-        },
-        [sourceLocation]
-    ).pipe(fs.createWriteStream(tarFilePath));
+    let source = sourceLocation.substring(1);
+    let current = process.cwd();
```

Comment:
The code doesn't restore the current folder

Reply:
If you restore the current dir before tar, how does tar.c know the correct item path? The behavior is very unusual if it works like you describe

Reply:
Oh you're right, forgot to check tar. Fixed!